meson: security/pam_appl.h header missing during compilation

The `security/pam_appl.h` header is missing during the compile step when using Meson. This should be properly located and required when running `meson setup`.

```
$ meson compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /usr/bin/ninja -C /var/home/jordan/Projects/util-linux/build
ninja: Entering directory `/var/home/jordan/Projects/util-linux/build'
[260/784] Compiling C object chfn.p/login-utils_auth.c.o
FAILED: chfn.p/login-utils_auth.c.o 
cc -Ichfn.p -I. -I.. -Iinclude -I../include -Ilibsmartcols -I../libsmartcols -I../libsmartcols/src -Ilibfdisk -I../libfdisk -I../libfdisk/src -I../libuuid/src -I../liblastlog2/src -Isys-utils -I../sys-utils -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O0 -g -D_GNU_SOURCE -fsigned-char -include /var/home/jordan/Projects/util-linux/build/config.h -fno-common -Waddress-of-packed-member -Wdiscarded-qualifiers -Wformat-security -Wimplicit-function-declaration -Wmissing-declarations -Wmissing-parameter-type -Wmissing-prototypes -Wnested-externs -Wno-missing-field-initializers -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wsign-compare -Wstrict-prototypes -Wtype-limits -Wuninitialized -Wunused-but-set-parameter -Wunused-but-set-variable -Wunused-parameter -Wunused-result -Wunused-variable -Wvla -Walloca -MD -MQ chfn.p/login-utils_auth.c.o -MF chfn.p/login-utils_auth.c.o.d -o chfn.p/login-utils_auth.c.o -c ../login-utils/auth.c
../login-utils/auth.c:11:10: fatal error: security/pam_appl.h: No such file or directory
   11 | #include <security/pam_appl.h>
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

--- a/meson.build
+++ b/meson.build
@@ -356,16 +356,13 @@ if not lib_crypt.found()
   lib_crypt = cc.find_library('crypt', required : get_option('build-sulogin'))
 endif
 
-lib_pam = cc.find_library('pam', required : get_option('build-login'))
-if not lib_pam.found()
-  lib_pam = cc.find_library('pam', required : get_option('build-chfn-chsh'))
-endif
-if not lib_pam.found()
-  lib_pam = cc.find_library('pam', required : get_option('build-su'))
-endif
-if not lib_pam.found()
-  lib_pam = cc.find_library('pam', required : get_option('build-runuser'))
-endif
+lib_pam = cc.find_library(
+  'pam',
+  disabler : true,
+  required : get_option('build-login').enabled() or \
+    get_option('build-chfn-chsh').enabled() or \
+    get_option('build-su').enabled() or \
+    get_option('build-runuser').enabled())
 if lib_pam.found()
   lib_pam_misc = cc.find_library('pam_misc')
   lib_pam = [lib_pam, lib_pam_misc]

